server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -592,13 +598,26 @@ public boolean isTranslogSyncNeeded() {
         return getTranslog().syncNeeded();
     }
 
+    private AsyncIOProcessor<Translog.Location> createTranslogSyncProcessor(Logger logger, ThreadContext threadContext) {
+        return new AsyncIOProcessor<>(logger, 1024, threadContext) {
+            @Override
+            protected void write(List<Tuple<Translog.Location, Consumer<Exception>>> candidates) throws IOException {
+                try {
+                    final boolean synced = translog.ensureSynced(candidates.stream().map(Tuple::v1));
+                    if (synced) {
+                        revisitIndexDeletionPolicyOnTranslogSynced();
+                    }
+                } catch (IOException ex) { // if this fails we are in deep shit - fail the request
+                    logger.debug("failed to sync translog", ex);
+                    throw ex;
+                }
+            }
+        };
+    }
```

Comment:
Is this moved from `org.elasticsearch.index.shard.IndexShard#createTranslogSyncProcessor`? 
If so I wonder why `catch (AlreadyClosedException ex)` was removed?